github-extension/src/browser/github-quick-open-service.ts

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+
+import { injectable, inject } from "inversify";
+import { MessageService } from '@theia/core/lib/common/message-service';
+import { QuickOpenItem, QuickOpenModel, QuickOpenMode } from "@theia/core/lib/browser/quick-open/quick-open-model";
+import { QuickOpenService, QuickOpenOptions } from '@theia/core/lib/browser/quick-open/quick-open-service';
+import { open, OpenerService } from "@theia/core/lib/browser";
+import { propertiesUri, GitHubProperties } from './github-properties';
+import { GithubService } from "../common/github-service";
+import { Repository } from "../common/github-model";
+import { Git } from "@theia/git/lib/common";
+import URI from "@theia/core/lib/common/uri";
+import { WorkspaceService } from "@theia/workspace/lib/browser/workspace-service";
+import { FileSystem, FileStat } from "@theia/filesystem/lib/common/filesystem";
+import { SelectionService } from "@theia/core/lib/common";
+import { StatusBar, StatusBarAlignment } from '@theia/core/lib/browser/status-bar/status-bar';
+
+@injectable()
+export class GitHubQuickOpenService {
+
+    private rootUri: URI | undefined;
+    private gitUrlRegex = /(?:git|ssh|https?|git@[-\w.]+):(\/\/)?(.*?)(\.git)(\/?|#[-\d\w._]+?)$/;
+
+    constructor(
+        @inject(MessageService) protected readonly messageService: MessageService,
+        @inject(QuickOpenService) protected readonly quickOpenService: QuickOpenService,
+        @inject(OpenerService) protected readonly openerService: OpenerService,
+        @inject(GitHubProperties) protected readonly gitHubProperties: GitHubProperties,
+        @inject(GithubService) protected readonly gitHubService: GithubService,
+        @inject(Git) protected readonly git: Git,
+        @inject(WorkspaceService) protected readonly workspaceService: WorkspaceService,
+        @inject(FileSystem) protected readonly fileSystem: FileSystem,
+        @inject(SelectionService) protected readonly selectionService: SelectionService,
+        @inject(StatusBar) protected readonly statusBar: StatusBar
+    ) {
+        workspaceService.root.then(root => {
+            if (root) {
+                this.rootUri = new URI(root.uri);
+            }
+        });
+    }
+
+    openConfigurationFile() {
+        open(this.openerService, propertiesUri);
+    }
+
+    async cloneRepository(): Promise<void> {
+        const __this = this;
```

Comment:
Why you need this construction???

Reply:
Because the code from lambdas like in https://github.com/eclipse/che-theia-github-plugin/pull/2/files#diff-28ed9c599a71f8aa641661b127a7d45dR78 doesn't have access to the current class members by keyword `this`, so we need to get access to the class members through the different variables that refers to the `this` object as I've correctly understood from the specification of TypeScript.